src/sections/new-collection/collection-form/useSubmitCollection.ts

```diff
@@ -0,0 +1,82 @@
+import { useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { WriteError } from '@iqss/dataverse-client-javascript'
+import { createCollection } from '../../../collection/domain/useCases/createCollection'
+import { CollectionRepository } from '../../../collection/domain/repositories/CollectionRepository'
+import { CollectionDTO } from '../../../collection/domain/useCases/DTOs/CollectionDTO'
+import { CollectionFormData, CollectionFormValuesOnSubmit } from '.'
+import { Route } from '../../Route.enum'
+import { JSDataverseWriteErrorHandler } from '../../../shared/helpers/JSDataverseWriteErrorHandler'
+
+export enum SubmissionStatus {
+  NotSubmitted = 'NotSubmitted',
+  IsSubmitting = 'IsSubmitting',
+  SubmitComplete = 'SubmitComplete',
+  Errored = 'Errored'
+}
+
+type UseSubmitCollectionReturnType =
+  | {
+      submissionStatus:
+        | SubmissionStatus.NotSubmitted
+        | SubmissionStatus.IsSubmitting
+        | SubmissionStatus.SubmitComplete
+      submitForm: (formData: CollectionFormData) => void
+      submitError: null
+    }
+  | {
+      submissionStatus: SubmissionStatus.Errored
+      submitForm: (formData: CollectionFormData) => void
+      submitError: string
+    }
+
+export function useSubmitCollection(
+  collectionRepository: CollectionRepository,
+  onSubmitErrorCallback: () => void
+): UseSubmitCollectionReturnType {
+  const navigate = useNavigate()
+
+  const [submissionStatus, setSubmissionStatus] = useState<SubmissionStatus>(
+    SubmissionStatus.NotSubmitted
+  )
+  const [submitError, setSubmitError] = useState<string | null>(null)
+
+  const submitForm = (formData: CollectionFormValuesOnSubmit): void => {
+    setSubmissionStatus(SubmissionStatus.IsSubmitting)
+
+    const newCollection: CollectionDTO = {
+      name: formData.name,
+      alias: formData.alias,
+      type: formData.type,
+      contacts: formData.contacts.map((contact) => contact.value)
+    }
+
+    const hostCollection = formData.hostCollection
+
+    createCollection(collectionRepository, newCollection, hostCollection)
+      .then(() => {
+        setSubmitError(null)
+        setSubmissionStatus(SubmissionStatus.SubmitComplete)
+
+        navigate(`${Route.COLLECTIONS}?id=${newCollection.alias}`, {
+          state: { created: true }
+        })
```

Comment:
Can we add a 'scroll to top' when navigating to the new collection (and new dataset)?  When I tested, I thought it was an error at first because I had scrolled to the bottom to submit the form, and the bottom view of the new collection page didn't show the sucess message or page title. It might help users to always see the top of the page.